samples/drivers/soc_flash_stm32h7/README.rst

```diff
@@ -0,0 +1,81 @@
+.. zephyr:code-sample:: soc-flash-stm32h7
+   :name: STM32H7 SoC flash
+   :relevant-api: flash_interface flash_area_api
+
+   Use the flash API to interact with the SoC flash.
+
+Overview
+********
+
+This sample demonstrates using the :ref:`Flash API <flash_api>` on an SoC internal flash.
+The sample uses :ref:`Flash map API <flash_map_api>` to obtain device for flash, using
+DTS node label, and then directly uses :ref:`Flash API <flash_api>` to perform
+flash operations.
+
+Within the sample, user may observe how read/write/erase operations are performed on a
+device, and how to first check whether device is ready for operation.
+
+Limitations
+***********
+Only tested when either M4 or M7 is running, not when both are running (co-existence).
+Each M4 and M7 owns Bank1 and Bank2 of the flash device respectively.
```

Comment:
Looks like a pretty strong limitation. Is that still valid ?

Reply:
Yes. I don't know why they can't work together as there is a semaphore used for serializing access to common register per my understanding. Also I see the write very slow from M4 based on upstream main branch, It took about 6 minutes to write a 256KiB block? Isn't that abnormal? What could be making it slow? Is it possible for someone to test this independently on a STM32H747i-disco?  We have a custom board based on STM32H755 and I see the write is faster close what is on M7. What could make STM32H747i-disco slower?

Reply:
I've been able to reproduce this issue and that looks odd indeed.